### PR TITLE
New docs site

### PR DESCRIPTION
**Pull request by djmitche**
I'm working on rewriting the docs site to use mdbook.  I've done a bit of work already to make the playground work and to render examples nicely.

The one thing this doesn't do is allow linking to the playground.  For the moment, I think it's OK to leave that out for now.  We can add such support later.

### CODE REVIEW

**Comment by djmitche** [comment]
Hm, the dark/light theming is kind of a mess here.  I'll keep hacking on it.

**Comment by djmitche** [comment]
OK, here's an updated version.  I pretty much left the existing mdbook themes in place, but added a touch of "JSON-e green" here and there.  It's deployed on https://json-e.js.org/ now.

A neat side benefit here, is that the library is now usable from a CDN such as jsdelivr, as it builds js/dist/index.js that rolls up the entire implementation into one file.

File: TODO.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* eliminate deps with https://cdn.jsdelivr.net/npm/json-e@4.4.3 and similar
```
Comment (jwhitlock) [inline TODO.txt:1]: this file could be `TODO.md`, looks like Markdown

**Review by jwhitlock — APPROVED** [review]
Nice work @djmitche!

File: TODO.txt (hunk 1 above)
Comment (djmitche) [reply TODO.txt:1]: Oh, how did I miss that -- that's just my notes.  I'll rebase that out and merge.

**Pull request merged**